.travis, build, vendor: autodelete old unstable archives

karalabe opened the pull request:
This PR adds a build step to travis to auto-delete **unstable** archives older than 14 days (our regular release schedule) from Azure via `ci.go purge`.

The PR also pulls in the latest Azure storage code, also switching over from the old import path (`github.com/Azure/azure-sdk-for-go`) to the new split one (`github.com/Azure/azure-storage-go`).

---

Btw, we'll need to enable daily cron jobs on travis for master **after** merging this in.

fjl commented:
Do we really need this on every build? It would be nice to set up a Travis Cron Job for it.

karalabe commented:
Less moving components this way. A cron job is fine too, but that's one more concept that could go wrong. I'll look into it.

karalabe commented:
@fjl I've added the cron-job detection code on travis, appveyor and via manual flags. I also added that `maybeSkipArchive` skips if it's a cron job, whilst the purging only runs if it **is** a cron job. I couldn't get Travis not to list the build job at all during PRs, since even whitelisting only master, every PR against master is considered to match it. PTAL
